misc/random.c

```diff
@@ -44,6 +46,13 @@ void mp_rand_seed(uint64_t seed)
 #ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
     seed = 42;
 #endif
+
+    if(seed == 0)
+    {
+        seed = av_get_random_seed(); // 32bit
+        seed = (seed << 32) ^ av_get_random_seed();
```

Comment:
Why `xor`? Lower bits will be zero after shift anyway, so doing `or` `(seed << 32) | av_get_random_seed()` should be fine.

Reply:
xor is an automatism for me when dealing with random bits. | is fine as long as av_get_random_seed() never changes to uint64. Saw your suggested change too late just now, sorry about that.